test/unit/com/unifina/domain/security/IntegrationKeySpec.groovy

```diff
@@ -1,16 +1,26 @@
 package com.unifina.domain.security
 
-import spock.lang.Specification
-
-class IntegrationKeySpec extends Specification {
+import com.unifina.BeanMockingSpecification
+import com.unifina.service.EthereumIntegrationKeyService
+import grails.test.mixin.Mock
+import grails.test.mixin.TestFor
+@TestFor(IntegrationKey)
+@Mock([SecUser])
+class IntegrationKeySpec extends BeanMockingSpecification {
+	def setup() {
+		EthereumIntegrationKeyService ethereumIntegrationKeyService = new EthereumIntegrationKeyService()
+		mockBean(EthereumIntegrationKeyService.class, ethereumIntegrationKeyService)
```

Comment:
Best practice is to mock collaborators in unit tests. Here it would mean setting the bean to a `Mock` instead of the real service:

```
EthereumIntegrationKeyService ethereumIntegrationKeyService

def setup() {
    ethereumIntegrationKeyService = mockBean(EthereumIntegrationKeyService, Mock(EthereumIntegrationKeyService))
}
```

then since you're mocking the decryption, you are free to set the `IntegrationKey#privateKey` to something easier to read

```
json: '{"address": "0xa3d1f77acff0060f7213d7bf3c7fec78df847de1", "privateKey": "encrypted-private-key"}'
```

then in your test:

```
when:
Map result = key.toMap()

then:
// Assert decryptPrivateKey(key) was called, and define the return value
1 * ethereumIntegrationKeyService.decryptPrivateKey(key) >> "decrypted-private-key"
// Check that the returned structure is correct
result.json == [address: "0xa3d1f77acff0060f7213d7bf3c7fec78df847de1", privateKey: "decrypted-private-key"]
```